lib/workers/repository/onboarding/branch/rebase.ts

```diff
@@ -0,0 +1,76 @@
+import is from '@sindresorhus/is';
+import { GlobalConfig } from '../../../../config/global';
+import type { RenovateConfig } from '../../../../config/types';
+import { logger } from '../../../../logger';
+import { scm } from '../../../../modules/platform/scm';
+import { getFile } from '../../../../util/git';
+import { toSha256 } from '../../../../util/hasha';
+import { OnboardingState, defaultConfigFile } from '../common';
+import { OnboardingCommitMessageFactory } from './commit-message';
+import { getOnboardingConfigContents } from './config';
+
+export async function rebaseOnboardingBranch(
+  config: RenovateConfig,
+  previousConfigHash?: string
+): Promise<string | null> {
+  logger.debug('Checking if onboarding branch needs rebasing');
+  const configFile = defaultConfigFile(config);
+  const existingContents =
+    (await getFile(configFile, config.onboardingBranch)) ?? '';
+  const currentConfigHash = toSha256(existingContents);
+  const contents = await getOnboardingConfigContents(config, configFile);
+
+  if (config.onboardingRebaseCheckbox) {
+    handleOnboardingManualRebase(previousConfigHash, currentConfigHash);
+  }
+
+  if (contents === existingContents) {
+    logger.debug('Onboarding branch is up to date');
+    return null;
+  }
```

Comment:
Should this code be moved to a different function?

Reply:
Not just for the sake of it. If you have a good reason, then sure